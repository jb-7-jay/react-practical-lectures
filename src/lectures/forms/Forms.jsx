import React, { useState, useEffect } from 'react';

const Forms = () => {

    const [firstname, setFirstname] = useState("");

    useEffect(() => {
        setFirstname('hello')
    }, [])

    return (
        <div className="center-of-screen">
            <div>
                <input 
                    className="an-16 demi-bold-text pa10" 
                    type="text" 
                    value={ firstname } 
                    onChange={(e) => setFirstname(e.target.value)}
                />
                <div className="an-16 demi-bold-text redcolor--text pt20">
                    Your FirstName is:  { firstname }
                </div>
            </div>
        </div>
    )
}

export default Forms;