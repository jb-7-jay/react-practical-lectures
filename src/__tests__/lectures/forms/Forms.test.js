import React from "react";
import Forms from "lectures/forms/Forms";
import TestRenderer from "react-test-renderer";
import { create, act } from "react-test-renderer";

describe("Check Forms in Lectures", () => {
  test("check state value", () => {
    let tree = create(<Forms />);
    console.log("treetreetreetree", tree);
    let instance = tree.getInstance();
    console.log("tree.props", tree.toJSON().props);
    // expect(instance.state.firstname).toBe("hello");
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
