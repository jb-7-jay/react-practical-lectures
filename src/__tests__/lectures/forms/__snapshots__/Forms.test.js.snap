// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Forms in Lectures check state value 1`] = `
<div
  className="center-of-screen"
>
  <div>
    <input
      className="an-16 demi-bold-text pa10"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div
      className="an-16 demi-bold-text redcolor--text pt20"
    >
      Your FirstName is:  
      
    </div>
  </div>
</div>
`;
